src/coreclr/gc/gc.cpp

```diff
@@ -11392,7 +11393,10 @@ heap_segment* gc_heap::get_free_region (int gen_number, size_t size)
             region = free_regions[huge_free_region].unlink_smallest_region (size);
             if (region == nullptr)
             {
-                ASSERT_HOLDING_SPIN_LOCK(&gc_lock);
+                if (settings.pause_mode != pause_no_gc)
+                {
+                    ASSERT_HOLDING_SPIN_LOCK(&gc_lock);
```

Comment:
You may want to add a comment here explaining why this is thread safe in the no-gc case.